cpp/example_code/s3-crt/s3-crt-demo.cpp

```diff
@@ -0,0 +1,227 @@
+
+    Aws::S3Crt::Model::PutObjectRequest request;
+    request.SetBucket(bucketName);
+    request.SetKey(objectKey);
+    std::shared_ptr<Aws::IOStream> bodyStream = Aws::MakeShared<Aws::FStream>(ALLOCATION_TAG, fileName.c_str(), std::ios_base::in | std::ios_base::binary);
+    if (!bodyStream->good()) {
+        std::cout << "Failed to open file: \"" << fileName << "\"." << std::endl << std::endl;
+        return false;
+    }
+    request.SetBody(bodyStream);
+
+    Aws::S3Crt::Model::PutObjectOutcome outcome = s3CrtClient.PutObject(request);
+
+    if (outcome.IsSuccess()) {
+        std::cout << "Object added." << std::endl << std::endl;
+
+        return true;
+    }
+    else {
+        std::cout << "PutObject error:\n" << outcome.GetError() << std::endl << std::endl;
+
+        return false;
+    }
+}
+
+// Get the Amazon S3 object from the bucket.
+bool GetObject(const Aws::S3Crt::S3CrtClient& s3CrtClient, const Aws::String& bucketName, const Aws::String& objectKey) {
+
+    std::cout << "Getting object: \"" << objectKey << "\" from bucket: \"" << bucketName << "\" ..." << std::endl;
+
+    Aws::S3Crt::Model::GetObjectRequest request;
+    request.SetBucket(bucketName);
+    request.SetKey(objectKey);
+
+    Aws::S3Crt::Model::GetObjectOutcome outcome = s3CrtClient.GetObject(request);
+
+    if (outcome.IsSuccess()) {
+        std::cout << "Object content: " << outcome.GetResult().GetBody().rdbuf() << std::endl << std::endl;
+
+        return true;
+    }
+    else {
+        std::cout << "GetObject error:\n" << outcome.GetError() << std::endl << std::endl;
+
+        return false;
+    }
+}
+
+// Delete the Amazon S3 object from the bucket.
+bool DeleteObject(const Aws::S3Crt::S3CrtClient& s3CrtClient, const Aws::String& bucketName, const Aws::String& objectKey) {
+
+    std::cout << "Deleting object: \"" << objectKey << "\" from bucket: \"" << bucketName << "\" ..." << std::endl;
+
+    Aws::S3Crt::Model::DeleteObjectRequest request;
+    request.SetBucket(bucketName);
+    request.SetKey(objectKey);
+
+    Aws::S3Crt::Model::DeleteObjectOutcome outcome = s3CrtClient.DeleteObject(request);
+
+    if (outcome.IsSuccess()) {
+        std::cout << "Object deleted." << std::endl << std::endl;
+
+        return true;
+    }
+    else {
+        std::cout << "DeleteObject error:\n" << outcome.GetError() << std::endl << std::endl;
+
+        return false;
+    }
+}
+
+// 1. List all buckets under the account
+// 2. Create an Amazon S3 bucket
+// 3. Put an object to the bucket
+// 4. Get the object
+// 5. Delete the object
+// 6. Delete the bucket
+int main(int argc, char* argv[]) {
+
+    Aws::SDKOptions options;
+
+    // Override default log level for AWS common runtime libraries to prevent from being overwhelmed by logs from them.
+    options.loggingOptions.crt_logger_create_fn = []() {
+        return Aws::MakeShared<Aws::Utils::Logging::DefaultCRTLogSystem>(ALLOCATION_TAG, Aws::Utils::Logging::LogLevel::Warn);
+    };
+
+#if 0
```

Comment:
```if 0```?

Reply:
I could comment out those lines as well.

Reply:
Why we release commented code?